File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        already_seen = {}

        for i, v in enumerate(nums):
            d = target - v
            if d in already_seen:
                return [already_seen[d], i]
            
            already_seen[v] = i

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.cpp
class Solution {
   public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int L = 0, curSum = 0;
        int length = INT_MAX;

        for (int R = 0; R < nums.size(); R++) {
            curSum += nums[R];
            while (curSum >= target) {
                length = min(R - L + 1, length);
                curSum -= nums[L];
                L++;
            }
        }
        return length == INT_MAX ? 0 : length;
    }
};